File: lib/core/repositories/squads_repository.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../models/squad_definition_model.dart';
import '../supabase/supabase_client.dart';

final squadsProvider = FutureProvider<List<SquadDefinitionModel>>((ref) async {
  final data =
      await supabase.from('squad_definitions').select().order('created_at');
  return (data as List).map((e) => SquadDefinitionModel.fromJson(e)).toList();
});

final squadBySlugProvider =
    FutureProvider.family<SquadDefinitionModel?, String>((ref, slug) async {
  final data = await supabase
      .from('squad_definitions')
      .select()
      .eq('slug', slug)
      .maybeSingle();
  if (data == null) return null;
  return SquadDefinitionModel.fromJson(data);
});

File: src/core/repositories/squads.repository.ts
import { supabase } from "@/core/supabase/client";
import { squadDefinitionFromRow, type SquadDefinitionModel } from "@/core/models/squad-definition";

export async function fetchSquads(): Promise<SquadDefinitionModel[]> {
  const { data, error } = await supabase
    .from("squad_definitions")
    .select("*")
    .order("created_at");
  if (error) throw error;
  return data.map(squadDefinitionFromRow);
}

export async function fetchSquadBySlug(slug: string): Promise<SquadDefinitionModel | null> {
  const { data, error } = await supabase
    .from("squad_definitions")
    .select("*")
    .eq("slug", slug)
    .maybeSingle();
  if (error) throw error;
  return data ? squadDefinitionFromRow(data) : null;
}
